Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Update.ts

```diff
@@ -254,64 +255,87 @@ async function checkForInvalidReferences(
+  session: ClientSession,
+): Promise<UpdateResult> {
+  const { meadowlarkId, documentUuid, resourceInfo, documentInfo, edfiDoc, validateDocumentReferencesExist, security } =
+    updateRequest;
+
+  if (validateDocumentReferencesExist) {
+    const invalidReferenceResult: UpdateResult | null = await checkForInvalidReferences(
+      updateRequest,
+      mongoCollection,
+      session,
+    );
+    if (invalidReferenceResult !== null) {
+      return invalidReferenceResult;
+    }
+  }
+
+  const document: MeadowlarkDocument = meadowlarkDocumentFrom(
+    resourceInfo,
+    documentInfo,
+    documentUuid,
+    meadowlarkId,
+    edfiDoc,
+    validateDocumentReferencesExist,
+    security.clientId,
+  );
+
+  if (resourceInfo.allowIdentityUpdates) {
+    return updateAllowingIdentityChange(document, updateRequest, mongoCollection, session);
+  }
+  return updateDisallowingIdentityChange(document, updateRequest, mongoCollection, session);
+}
+
 /**
  * Takes an UpdateRequest and MongoClient from the BackendFacade and performs an update by documentUuid
  * and returns the UpdateResult.
  */
 export async function updateDocumentById(updateRequest: UpdateRequest, client: MongoClient): Promise<UpdateResult> {
-  const {
-    meadowlarkId,
-    documentUuid,
-    resourceInfo,
-    documentInfo,
-    edfiDoc,
-    validateDocumentReferencesExist,
-    traceId,
-    security,
-  } = updateRequest;
+  const { documentUuid, traceId } = updateRequest;
   Logger.info(`${moduleName}.updateDocumentById ${documentUuid}`, traceId);
 
   const mongoCollection: Collection<MeadowlarkDocument> = getDocumentCollection(client);
   const session: ClientSession = client.startSession();
   let updateResult: UpdateResult = { response: 'UNKNOWN_FAILURE' };
 
   try {
-    await session.withTransaction(async () => {
-      if (validateDocumentReferencesExist) {
-        const invalidReferenceResult: UpdateResult | null = await checkForInvalidReferences(
-          updateRequest,
-          mongoCollection,
-          session,
-        );
-        if (invalidReferenceResult !== null) {
-          updateResult = invalidReferenceResult;
-          return; // exit transaction block
-        }
-      }
-
-      const document: MeadowlarkDocument = meadowlarkDocumentFrom(
-        resourceInfo,
-        documentInfo,
-        documentUuid,
-        meadowlarkId,
-        edfiDoc,
-        validateDocumentReferencesExist,
-        security.clientId,
-      );
-
-      if (resourceInfo.allowIdentityUpdates) {
-        updateResult = await updateAllowingIdentityChange(document, updateRequest, mongoCollection, session);
-      } else {
-        updateResult = await updateDisallowingIdentityChange(document, updateRequest, mongoCollection, session);
-      }
-
-      if (updateResult.response !== 'UPDATE_SUCCESS') {
-        await session.abortTransaction();
-      }
-    });
+    const numberOfRetries: number = Config.get('MONGODB_MAX_NUMBER_OF_RETRIES') ?? 1;
```

Comment:
This is why we want that default value to be set in the `Config` module - so that we don't need to repeat ourselves every time we access the config value.